src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/test/CustomITEmbeddingProvider.java

```diff
@@ -60,13 +59,10 @@ public CustomITEmbeddingProvider(int dimension) {
 
   @Override
   public Uni<Response> vectorize(
-      int batchId,
-      List<String> texts,
-      Optional<String> apiKey,
-      EmbeddingRequestType embeddingRequestType) {
+      int batchId, List<String> texts, String apiKey, EmbeddingRequestType embeddingRequestType) {
     List<float[]> response = new ArrayList<>(texts.size());
     if (texts.size() == 0) return Uni.createFrom().item(Response.of(batchId, response));
-    if (!apiKey.isPresent() || !apiKey.get().equals(TEST_API_KEY))
+    if (null == apiKey || !apiKey.equals(TEST_API_KEY))
```

Comment:
Fine as is, but can simplify as:

    if (!TEST_API_KEY.equals(apiKey)) {

since `String.equals(null)` is valid (returns `false`)